File: BOJ/#4673 셀프 넘버/셀프 넘버.cpp
#include <iostream>
#include <vector>
using namespace std;

#define SIZE 10001

int main() {
    vector<bool> isSelf(SIZE, true);

    for(int i=1; i<SIZE; i++) {
        int sum=0, n=i;
        while(n) {
            sum += n%10;
            n /= 10;
        }
        sum += i;

        if(sum < SIZE) isSelf[sum] = false;
    }

    for(int i=1; i<SIZE; i++)
        if(isSelf[i]) printf("%d\n", i);
}

File: BOJ/#4673 셀프 넘버/셀프 넘버.py
SIZE = 10001

isSelf = [True for i in range(SIZE)]

for i in range(1, SIZE):
    sum,n = 0,i
    while n > 0:
        sum += n%10
        n //= 10
    sum += i

    if sum < SIZE: isSelf[sum] = False

for i in range(1, SIZE):
    if isSelf[i]: print(i)
